### PR TITLE
OCPBUGS-31497: create suitable role and roleBinding for csi-snapshot-webhook

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/OCPBUGS-31497

This PR:
- Sets ClusterRoles for both the controller and webhook to what is recommended upstream (see [controller](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml) and [webhook](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/webhook-example/rbac-snapshot-webhook.yaml) examples). Also grant these permissions to the operator so it can create these roles.
- Adds volumegroupsnapshot to the webhook config as demonstrated [upstream](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/webhook-example/admission-configuration-template)
- Moves the ClusterRoles and ClusterRoleBindings to StaticResourceController, so these get applied even when the VolumeGroupSnapshot API is not enabled.

When we backport this to fix the volumesnapshotclass RBAC failures on older versions, we'll also need some of the changes from https://github.com/openshift/cluster-csi-snapshot-controller-operator/commit/0d3d951ca9c5d2affb82442e958e7bd5852393bb -- specifically the part where we create the webhook service account and add it to the webhook deployment, and the assets/assets.go change where we include `rbac/*.yaml`.

/cc @openshift/storage @bertinatto @Phaow

### CODE REVIEW

**Comment by dobsonj** [comment]
The CI failures are legit
/hold

**Comment by dobsonj** [comment]
/unhold
See the latest [controller logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_cluster-csi-snapshot-controller-operator/202/pull-ci-openshift-cluster-csi-snapshot-controller-operator-master-e2e-gcp-csi/1774883715311210496/artifacts/e2e-gcp-csi/gather-extra/artifacts/pods/openshift-cluster-storage-operator_csi-snapshot-controller-54fdc874f4-fdxmk_snapshot-controller.log) and [webhook logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_cluster-csi-snapshot-controller-operator/202/pull-ci-openshift-cluster-csi-snapshot-controller-operator-master-e2e-gcp-csi/1774883715311210496/artifacts/e2e-gcp-csi/gather-extra/artifacts/pods/openshift-cluster-storage-operator_csi-snapshot-webhook-697848c8b6-z9d6c_webhook.log) from e2e-gcp-csi

File: manifests/05_operator_clusterrole.yaml [hunk 1]
```diff
@@ -82,24 +82,4 @@ rules:
...
-  - volumegroupsnapshotcontents
-  verbs:
-  - watch
-  - list
-  - get
-  - create
-  - delete
-  - patch
-  - update
+  - "*"
```
Comment (bertinatto) [inline manifests/05_operator_clusterrole.yaml:85]: We used to use wildcards in our roles, but were told to remove them and list the permissions explicitly

**Review by bertinatto — COMMENTED** [review]
Just a comment, otherwise LGTM.

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**